lib/puppet/util/windows/service.rb

```diff
@@ -0,0 +1,437 @@
+      open_service(service_name) do |service|
+        wait_for_service_pending(service, SERVICE_STOP_PENDING, SERVICE_STOPPED)
+        unless StartServiceW(service, 0, nil)
+          raise Puppet::Util::Windows::Error.new(_("Failed to start the service"))
+        end
+        wait_for_service_pending(service, SERVICE_START_PENDING, SERVICE_RUNNING)
+      end
+    end
+    module_function :start
+
+    # Use ControlService to send a stop signal to a windows service
+    #
+    # @param [:string] service_name name of the service to stop
+    def stop(service_name)
+      status = SERVICE_STATUS.new
+      open_service(service_name) do |service|
+        wait_for_service_pending(service, SERVICE_START_PENDING, SERVICE_RUNNING)
+        unless ControlService(service, SERVICE_CONTROL_STOP, status)
+          raise Puppet::Util::Windows::Error.new(_("Failed to send stop control to service"))
+        end
+        wait_for_service_pending(service, SERVICE_STOP_PENDING, SERVICE_STOPPED)
+      end
+    end
+    module_function :stop
+
+    # Query the state of a service using QueryServiceStatusEx
+    #
+    # @param [:string] service_name name of the service to query
+    # @return [string] the status of the service
+    def service_state(service_name)
+      status = nil
+      open_service(service_name) do |service|
+        status = query_status(service)
+      end
+      raise Puppet::Error.new(_("Something went wrong querying the service")) if status[:dwCurrentState].nil?
+      status[:dwCurrentState]
+    end
+    module_function :service_state
+
+    # Query the configuration of a service using QueryServiceConfigW
+    #
+    # @param [:string] service_name name of the service to query
+    # @return [QUERY_SERVICE_CONFIGW.struct] the configuration of the service
+    def service_config(service_name)
+      config = nil
+      open_service(service_name) do |service|
+        config = query_config(service)
+      end
+      config
+    end
+    module_function :service_config
+
+    def configure(service_name, configuration_updates)
+    end
+    module_function :configure
+
+    # @api private
+    # Opens a connection to the SCManager on windows then uses that
+    # handle to create a handle to a specific service in windows
+    # corresponding to service_name
+    #
+    # this function takes a block that executes within the context of
+    # the open service handler, and will close the service and SCManager
+    # handles once the block finishes
+    #
+    # @param [string] service_name the name of the service to open
+    # @yieldparam [:handle] service the windows native handle used to access
+    #   the service
+    def self.open_service(service_name)
+      scm = OpenSCManagerW(nil, nil, SC_MANAGER_ALL_ACCESS)
+      raise Puppet::Util::Windows::Error.new(_("Failed to open a handle to the service control manager")) if scm == FFI::Pointer::NULL_HANDLE
+      service = OpenServiceW(scm, service_name, SC_MANAGER_ALL_ACCESS)
```

Comment:
I think the reason the acceptance test is failing is b/c you need to cast `service_name` into a wide-string?